### ISSUE TITLE
[3.3.5] Boss: AQ20 Ossirian's crystals are not clickable.

### ISSUE THREAD

**Issue by stanislavkozlovski** [TrinityCore/TrinityCore#17315]
**Description**:
Ossirian has a 300% damage buff which you disable when you click on a crystal.
The crystals spawn but are not clickable as a player. I looked at the code and their ID is 180619 which in the DB is set as faction 35, which makes it clickable by all players. When spawned manually using .gobject add, the crystal is clickable. When the boss spawns it, it is not.
**Current behaviour**: Tell us what happens
The crystals spawn but are not clickable as a player.
The crystals are clickable as a GM and do what they are intended to do when clicked.

**Expected behaviour**: Tell us what should happen instead
They should be clickable to a player.

**Steps to reproduce the problem**:
Engage Ossirian and see for yourself.

**Branch(es)**: 
3.3.5
71081c2576571a6e6d5a16179e6729843094a882

**Comment by TrinityCoreBot** [comment]
This issue is not considered valid because of the following reasons:
- No proper commit hash (revision)

Please read http://bit.ly/tc-issuetracker-and-you . Thanks.

**Comment by malcrom** [comment]
I spawned AQ20. I would assume I have the wrong state set?

**Comment by Enether** [comment]
I have no idea what has gone wrong, I saw a commit saying "Objects are not clickable, WTF?!" and a bunch of others after that. I assume that it was fixed at one time. 
I looked through the code but could not see what was causing the issue. If somebody more experience could look at it it'd be great.

**Comment by Riztazz** [comment]
add Crystal->RemoveFromOwner(); when you're spawning that GO ;p

**Comment by tkrokli** [comment]
You mean something like this?
```diff
diff --git a/src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ossirian.cpp b/src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ossirian.cpp
index 14ccca0..99567f4 100644
--- a/src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ossirian.cpp
+++ b/src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ossirian.cpp
@@ -194,6 +194,7 @@ class boss_ossirian : public CreatureScript
                     {
                         CrystalGUID = Crystal->GetGUID();
                         ++CrystalIterator;
+                        Crystal->RemoveFromOwner();
                     }
                 }
             }
```

**Comment by Riztazz** [comment]
yes

**Comment by tkrokli** [comment]
@Enether : can you test this suggestion and report back with your results?

**Comment by Enether** [comment]
No sorry, I don't have anything set up and I switched to Linux so it's extremely inconvenient for me to test this out.

**Comment by KrypticKobra** [comment]
Tried running AQ20 last night and ran into this same issue.  Tried the fix suggested by Riztazz but doing so causes a failure to compile due to RemoveFromOwner() being a private function.  As a very dirty test I converted that function to public. After compiling the crystals are now clickable, and do seem to be both removing the boss' buff and applying the spell damage debuff to the boss correctly (I haven't run this instance in a Blizzard server so my knowledge is lacking here as to proper behavior).

Let me know if I can provide more information or testing.  Insight on a better fix for the private/public switch would be appreciated as I understand that is not good coding practice but wanted to test the fix anyway.
